### PR TITLE
Update slack to return empty iterable instead of a empty list

### PR DESCRIPTION
**Pull request by bnchrch**
## What

Source Slack is failing with 'list' object is not an iterator
https://airbytehq.sentry.io/issues/4508254133/?alert_rule_id=11478402&alert_type=issue&notification_uuid=1295203e-5b64-45d0-91d9-70b3377988a9&project=6527718

This seems to be related to Slack returning an empty list in one case
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-slack/source_slack/source.py#L213

Related to: https://github.com/airbytehq/airbyte/pull/31122

## How
Weve already updated the cdk to handle this for all connectors.

This PR
1. Updates the connector to use the new cdk
2. casts the array iterable on the slack side to be safe

### CODE REVIEW

**Review by roman-yermilov-gl — APPROVED** [review]

File: airbyte-integrations/connectors/source-slack/source_slack/source.py [hunk 1]
```diff
@@ -209,8 +209,9 @@ def read_records(
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
         if not stream_slice:
+            # return an empty iterable
             # this is done to emit at least one state message when no slices are generated
-            return []
+            return iter([])
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-slack/source_slack/source.py:214]:
```suggestion
            yield from []
```

Comment (erohmensing) [reply airbyte-integrations/connectors/source-slack/source_slack/source.py:214]: Both work, but suggestion based on how we handle this case in the CDK elsewhere (`http.py`, `simple_retriever.py`)

File: docs/integrations/sources/slack.md [hunk 2]
```diff
@@ -137,6 +137,7 @@ It is recommended to sync required channels only, this can be done by specifying
 
 | Version | Date       | Pull Request                                             | Subject                                                                             |
 |:--------|:-----------|:---------------------------------------------------------|:------------------------------------------------------------------------------------|
+| 0.3.4   | 2023-10-06 | [31134](https://github.com/airbytehq/airbyte/pull/31134) | Update CDK and remove non iterable return from records                              |
```
Comment (erohmensing) [inline docs/integrations/sources/slack.md:140]:
```suggestion
| 0.3.4   | 2023-10-06 | [31134](https://github.com/airbytehq/airbyte/pull/31134) | Update CDK and remove no-interator return from records                              |
```
.. I may have messed up the table with this suggestion so take it with a grain of salt lol

File: airbyte-integrations/connectors/source-slack/source_slack/source.py [hunk 3]
```diff
@@ -209,8 +209,9 @@ def read_records(
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
         if not stream_slice:
+            # return an empty iterable
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-slack/source_slack/source.py:212]:
```suggestion
            # return an empty iterator
```

**Review by erohmensing — APPROVED** [review]

File: docs/integrations/sources/slack.md (hunk 2 above)
Comment (bnchrch) [reply docs/integrations/sources/slack.md:140]: I.... do not like this change.

But! I loved all the others 😛

**Comment by airbyte-oss-build-runner** [comment]
##  source-slack test report (commit [`f7238eb65c`](https://github.com/airbytehq/airbyte/pull/31134/commits/f7238eb65c9419306b21281537c12be6a3a81172)) - ❌

⏲️  Total pipeline duration: 31.13s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-slack docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                   | ❌        |
| Code format checks                                           | ✅        |
| Validate metadata for source-slack                           | ✅        |
| Connector version semver check                               | ✅        |
| Connector version increment check                            | ✅        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bnchrch_source-slack-empty-list/1696616767/f7238eb65c9419306b21281537c12be6a3a81172/source-slack/0.3.4/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f7238eb65c9419306b21281537c12be6a3a81172)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-slack test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-slack test report (commit [`a5817fc75c`](https://github.com/airbytehq/airbyte/pull/31134/commits/a5817fc75c4d78290a8df6c605c70e0acfe2ac2a)) - ✅

⏲️  Total pipeline duration: 74mn52s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-slack docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                   | ✅        |
| Acceptance tests                                             | ✅        |
| Code format checks                                           | ✅        |
| Validate metadata for source-slack                           | ✅        |
| Connector version semver check                               | ✅        |
| Connector version increment check                            | ✅        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bnchrch_source-slack-empty-list/1696620363/a5817fc75c4d78290a8df6c605c70e0acfe2ac2a/source-slack/0.3.4/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:a5817fc75c4d78290a8df6c605c70e0acfe2ac2a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-slack test
```

**Pull request merged**